doc/developer/design/20230306_persist_mfp_pushdown.md

```diff
@@ -0,0 +1,272 @@
+readers generally need to fetch all the data in the collection as of the query time,
+even if they're only interested in a small subset.
+By pushing down filters into the Persist source,
+we can dramatically improve performance for certain subsets of queries,
+especially those that filter down to a narrow window of time.
+
+# Motivation
+[motivation]: #motivation
+
+Unlock append-only sources for customers via temporal filters. Right now, using an append-only `SOURCE` in Materialize is painful because `MATERIALIZED VIEW` restart times increase without bound as data accumulates. However, if there is a [temporal filter](https://materialize.com/docs/sql/patterns/temporal-filters/) to bound the size of the dataflow's input data (e.g. the last 30 days of data), we may be able to use the described optimization to similarly bound the restart times.
+
+This technique is not limited to temporal filters or sources. In fact, the optimization will be picked up anywhere we can apply it. However, we're hoping to focus on polishing the subset relevant to append-only sources in the first version of this shipped to customers.
+
+# Explanation
+[explanation]: #explanation
+
+## Schemas
+
+A `Schema` trait will be introduced to map data back and forth between Persist columns. Any Rust type
+that is to roundtrip through Persist must have a corresponding `Schema`.
+
+Each column in a `RelationDesc` is mapped to a column that persist understands: a name, a type, and override-able logic for computing aggregate statistics of columns of that data. The set of supported types is a subset of Apache [Arrow's data types], which is then internally mapped to [Parquet].
+
+[Arrow's data types]: https://docs.rs/arrow2/0.17.0/arrow2/datatypes/enum.DataType.html
+[parquet]: https://parquet.apache.org/docs/
+## Part Statistics
+
+When Persist writes a batch to Blob storage, it will compute basic statistics over its columns, when possible,
+and write this data alongside existing metadata in Consensus. To start, we will collect a column's
+non-null min and max values and null count. We may also track the min/max Row within the part.
+
+Statistics are optional: Persist may not be able to compute min/max values for all types, and we may need to trim statistics to keep metadata from getting too big.
+
+## Filtering
+
+Our compute layer pushes a `MapFilterProject` operator into the Persist source, which can filter and transform data before emitting it downstream.
+Before each batch part is fetched from Blob storage in `persist_source`, the source will use our new part statistics to analyze whether that MFP will filter out _all_ the rows in that part.
+If so, we get to skip fetching the part from Blob storage entirely.
+
+This requires that we have some way to figure out what an MFP's `MirScalarExpr`s might return given only basic information like the nullability and range of values. For more on how that can be done, see the section on [filtering parts].
```

Comment:
> some way to figure out what an MFP's `MirScalarExpr`s might return given only basic information like the nullability and range of values

nit: this sentence wasn't clear to me during the first read pass. I think more specifically you want to be able to do abstract interpretation / evaluation of the `predicates: Vec<MirScalarExpr>` in a `MapFilterProject` in order to discard parts that will not contribute results based only on their statistics.